consensus/bor/bor.go

```diff
@@ -1242,14 +1235,39 @@ func (c *Bor) CommitStates(
 		msg := getSystemMessage(common.HexToAddress(c.config.StateReceiverContract), data)
 
 		// apply message
-		if err := applyMessage(msg, state, header, c.chainConfig, chain); err != nil {
+		if err := applyMessage(msg, state, header, c.chainConfig, chCtx); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func validateEventRecord(eventRecord EventRecord, number uint64, from, to int64, chainID string) error {
+	_time, err := time.Parse(time.RFC3339, eventRecord.Time)
+	if err != nil {
+		return err
+	}
+	eventTime := _time.Unix()
+	if eventTime < from || eventTime > to {
+		return &InvalidStateReceivedError{
+			number,
+			time.Unix(int64(from), 0).Format(time.RFC3339),
+			time.Unix(int64(to), 0).Format(time.RFC3339),
+			&eventRecord}
+	}
```

Comment:
We should use `time.Before` API instead of parsing and formatting. https://golang.org/pkg/time/#example_Time_Before

Reply:
nice api